lib/mzgl/gl/api/OpenGLAPI.h

```diff
@@ -0,0 +1,337 @@
+		g.colorShader->loadFromString(STRINGIFY(
+
+			uniform mat4 mvp;
+			uniform lowp vec4 color;
+
+			in vec4 Position;
+			in lowp vec4 Color;
+
+			out lowp vec4 colorV;
+
+			void main(void) {
+				colorV	  = Color * color;
+				gl_Position = mvp * Position;
+			}
+
+		),
+
+		STRINGIFY(
+
+			in lowp vec4 colorV;
+			out vec4 fragColor;
+
+			void main(void) {
+				fragColor = colorV;
+			}
+
+		));
+
+		g.colorTextureShader = Shader::create(g);
+		g.colorTextureShader->loadFromString(STRINGIFY(
+
+			uniform mat4 mvp;
+
+			in vec4 Position;
+			in lowp vec2 TexCoord;
+			in lowp vec4 Color;
+
+			out lowp vec4 colorV;
+			out lowp vec2 texCoordV;
+			void main(void) {
+				colorV		= Color;
+				texCoordV	= TexCoord;
+				gl_Position = mvp * Position;
+			}
+
+		),
+STRINGIFY(
+
+			in lowp vec4 colorV;
+			in lowp vec2 texCoordV;
+			out vec4 fragColor;
+			uniform sampler2D myTextureSampler;
+
+			void main(void) {
+				fragColor = texture(myTextureSampler, texCoordV) * colorV;
+			}
+
+		));
+
+		g.fontShader = Shader::create(g);
+		g.fontShader->loadFromString(STRINGIFY(
+
+			uniform mat4 mvp;
+
+			in vec4 Position;
+			in vec2 TexCoord;
+
+			uniform lowp vec4 color;
+
+			out vec2 texCoordV;
+
+			void main() {
+				texCoordV	= TexCoord;
+				gl_Position = mvp * Position;
+			}
+
+		),
+STRINGIFY(
+
+			uniform lowp vec4 color;
+			in vec2 texCoordV;
+			out vec4 fragColor;
+			uniform sampler2D myTextureSampler;
+
+			void main() {
+				fragColor = color;
+				fragColor.a *= texture(myTextureSampler, texCoordV).a;
+			}
+
+		));
+		g.texShader = Shader::create(g);
+		g.texShader->loadFromString(STRINGIFY(
+
+			uniform mat4 mvp;
+
+			in vec4 Position;
+			in lowp vec2 TexCoord;
+			uniform lowp vec4 color;
+
+			out lowp vec4 colorV;
+			out lowp vec2 texCoordV;
+			void main(void) {
+				colorV		= color;
+				texCoordV	= TexCoord;
+				gl_Position = mvp * Position;
+			}
+
+		),
+	STRINGIFY(
+
+			in lowp vec4 colorV;
+			in lowp vec2 texCoordV;
+			out vec4 fragColor;
+			uniform sampler2D myTextureSampler;
+
+			void main(void) {
+				fragColor = texture(myTextureSampler, texCoordV) * colorV;
+			}
+
+		));
+		// clang-format on
+
+		g.nothingShader->isDefaultShader	  = true;
+		g.colorShader->isDefaultShader		  = true;
+		g.colorTextureShader->isDefaultShader = true;
+		g.texShader->isDefaultShader		  = true;
+		g.fontShader->isDefaultShader		  = true;
+	}
+
+	void setLineWidth(float f) override { glLineWidth(f); }
+	void setBlending(bool shouldBlend) override {
+		if (shouldBlend) {
+			glEnable(GL_BLEND);
+		} else {
+			glDisable(GL_BLEND);
+		}
+	}
+
+	void drawVerts(const std::vector<glm::vec2> &verts, Vbo::PrimitiveType type) override {
+		g.nothingShader->begin();
+
+		g.currShader->uniform("mvp", g.getMVP());
+		if (g.currShader->needsColorUniform) {
+			g.currShader->uniform("color", g.getColor());
+		}
+
+#ifndef MZGL_GL2
+		if (immediateVertexArray == 0) return;
+		glBindVertexArray(immediateVertexArray);
+#endif
```

Comment:
What mode do we use? I wonder if we should drop the other one in the cause of readability / maintainability

Reply:
one day we will drop GL2 but its the default at the moment - once the Sokol stuff works and we know it works on older Macs I'm happy to relinquish MZGL_GL2